packages/playground/src/components/dedicated_nodes_table.vue

```diff
@@ -0,0 +1,282 @@
+                  </a>
+                </li>
+              </ul>
+            </span>
+          </v-tooltip>
+        </template>
+
+        <template v-slot:expanded-row="{ columns, item }">
+          <node-details :item="item" :columns-len="columns.length"></node-details>
+        </template>
+      </v-data-table-server>
+    </v-card>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { NodeStatus } from "@threefold/gridproxy_client";
+import { onMounted, ref, watch } from "vue";
+import type { VDataTable } from "vuetify/labs/VDataTable";
+import { VDataTableServer } from "vuetify/labs/VDataTable";
+
+import { gridProxyClient } from "../clients";
+import { useProfileManager } from "../stores";
+import { getGrid, loadBalance } from "../utils/grid";
+import toTeraOrGigaOrPeta from "../utils/toTeraOrGegaOrPeta";
+
+const headers: VDataTable["headers"] = [
+  { title: "Node ID", key: "nodeId" },
+  { title: "Location", key: "location.country", sortable: false },
+  { title: "CRU", key: "total_resources.cru" },
+  { title: "MRU", key: "total_resources.mru", value: item => toTeraOrGigaOrPeta(item.total_resources.mru) },
+  { title: "SRU", key: "total_resources.sru", value: item => toTeraOrGigaOrPeta(item.total_resources.sru) },
+  { title: "HRU", key: "total_resources.hru", value: item => toTeraOrGigaOrPeta(item.total_resources.hru) },
+  { title: "GPU", key: "num_gpu" },
+  {
+    title: "Price (USD)",
+    key: "price",
+  },
+  {
+    title: "Actions",
+    key: "actions",
+    sortable: false,
+  },
+];
+const profileManager = useProfileManager();
+const pageSize = ref(10);
+const page = ref(1);
+const expanded = ref([]);
+const tabs = [{ label: "Rentable" }, { label: "Rented" }, { label: "Mine" }];
+const activeTab = ref(0);
+const loading = ref(false);
+const nodes = ref<any[]>();
+const nodesCount = ref(0);
+const filterInputs = ref<DedicatedNodeFilters>(DedicatedNodeInitializer);
+const isValidForm = ref<boolean>(false);
+
+const tabParams = {
+  0: {
+    rentable: true,
+    status: NodeStatus.Up,
+    retCount: true,
+  },
+  1: {
+    rented: true,
+    status: NodeStatus.Up,
+    retCount: true,
+  },
+  2: {
+    rented: true,
+    rentedBy: profileManager.profile?.twinId,
+    retCount: true,
+  },
+};
+
+onMounted(async () => {
+  await loadData();
+});
+
+async function loadData() {
+  const params = tabParams[activeTab.value as keyof typeof tabParams];
+
+  if (!params) {
+    return;
+  }
+
+  loading.value = true;
+  try {
+    const data = await gridProxyClient.nodes.list({
+      ...params,
+      size: pageSize.value,
+      page: page.value,
+      freeSru: filterInputs.value.total_sru.value ? +filterInputs.value.total_sru.value : 0,
+      freeHru: filterInputs.value.total_hru.value ? +filterInputs.value.total_hru.value : 0,
+      freeMru: filterInputs.value.total_mru.value ? +filterInputs.value.total_mru.value : 0,
+      totalCru: filterInputs.value.total_cru.value ? +filterInputs.value.total_cru.value : 0,
```

Comment:
is it possible to set the filter to an empty string or `undefined` in case the filter is not required cause the query would take much more time when they are set to 0